android/src/main/kotlin/com/adyen/checkout/flutter/dropIn/advanced/AdvancedDropInService.kt

```diff
@@ -45,8 +55,37 @@ class AdvancedDropInService : DropInService(), LifecycleOwner {
         }
     }
 
-    override fun onBalanceCheck(paymentComponentState: PaymentComponentState<*>) =
-        onPaymentComponentState(paymentComponentState)
+    override fun onBalanceCheck(paymentComponentState: PaymentComponentState<*>) {
+        try {
+            setBalanceCheckObserver()
+            val data = PaymentComponentData.SERIALIZER.serialize(paymentComponentState.data)
+            DropInBalanceCheckPlatformMessenger.sendResult(data)
+        } catch (exception: Exception) {
+            sendResult(
+                DropInServiceResult.Error(
+                    errorDialog = null,
+                    reason = exception.message,
+                    dismissDropIn = true
+                )
+            )
+        }
+    }
+
+    override fun onOrderRequest() {
+        setOrderRequestObserver()
+        DropInOrderRequestPlatformMessenger.sendResult("onOrderRequest")
```

Comment:
What is the use case of passing the "onOrderRequest" here?

Reply:
Just for triggering the observer that listens for the "onOrderRequest".